modules/graphql_core/src/Plugin/GraphQL/Fields/Routing/LanguageSwitchLinks.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Drupal\graphql_core\Plugin\GraphQL\Fields\Routing;
+
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Url;
+use Drupal\graphql\Plugin\GraphQL\Fields\FieldPluginBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Youshido\GraphQL\Execution\ResolveInfo;
+
+/**
+ * @GraphQLField(
+ *   id = "url_language_switch_links",
+ *   secure = true,
+ *   name = "languageSwitchLinks",
+ *   multi = true,
+ *   type = "LanguageSwitchLink",
+ *   types = {"Url"}
+ * )
+ */
+class LanguageSwitchLinks extends FieldPluginBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The language manager.
+   *
+   * @var \Drupal\Core\Language\LanguageManagerInterface
+   */
+  protected $languageManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static($configuration, $plugin_id, $plugin_definition, $container->get('language_manager'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $pluginId, $pluginDefinition, LanguageManagerInterface $languageManager) {
+    $this->languageManager = $languageManager;
+    parent::__construct($configuration, $pluginId, $pluginDefinition);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function resolveValues($value, array $args, ResolveInfo $info) {
```

Comment:
This should probably issue a sub-request. Right? /cc @pmelab

Reply:
Yeah, probably. Because `ConfigurableLanguageManager::getLanguageSwitchLinks()` passes current request to negotiators.